Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.BaseCollection.cs

```diff
@@ -0,0 +1,108 @@
+
+using System;
+using System.Collections;
+using System.Diagnostics.Contracts;
+
+namespace System.Windows.Forms
+{
+    /// <summary>
+    /// Provides the base functionality for creating data-related collections in the <see cref="N:System.Windows.Forms"/> namespace.
+    /// </summary>
+    public class BaseCollection : MarshalByRefObject // ICollection, IEnumerable
+    {
+        /// <summary>
+        /// Gets the total number of elements in the collection.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The total number of elements in the collection.
+        /// </returns>
+        
+        public virtual int Count
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<int>() >= 0);
+                return default(int);
+            }
+        }
+
+        // <summary>
+        // Gets a value indicating whether the collection is read-only.
+        // </summary>
+        // 
+        // <returns>
+        // This property is always false.
+        // </returns>
+        // public bool IsReadOnly { get; }
+        
+        /// <summary>
+        /// Gets a value indicating whether access to the <see cref="T:System.Collections.ICollection"/> is synchronized.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// This property always returns false.
+        /// </returns>
+        public bool IsSynchronized
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<bool>() == false);
+                return false;
+            }
+        }
+
+        // <summary>
+        // Gets an object that can be used to synchronize access to the <see cref="T:System.Windows.Forms.BaseCollection"/>.
+        // </summary>
+        // 
+        // <returns>
+        // An object that can be used to synchronize the <see cref="T:System.Windows.Forms.BaseCollection"/>.
+        // </returns>
+        public object SyncRoot
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<object>() != null);
+                return default(object);
+            }
+        }
+        
+        /// <summary>
+        /// Copies all the elements of the current one-dimensional <see cref="T:System.Array"/> to the specified one-dimensional <see cref="T:System.Array"/> starting at the specified destination <see cref="T:System.Array"/> index.
+        /// </summary>
+        /// <param name="ar">The one-dimensional <see cref="T:System.Array"/> that is the destination of the elements copied from the current Array. </param><param name="index">The zero-based relative index in <paramref name="ar"/> at which copying begins. </param>
+        public void CopyTo(Array ar, int index)
+        {
+            Contract.Requires(ar != null);
+            Contract.Requires(index >= 0);
```

Comment:
It seems that in previous file one contract statement was used: `Contract.Requires(index >= 0 && index <= Count)`.

Should we use consistent approach/